frontend/src/components/Map/DetailsMap/DetailsMap.tsx

```diff
@@ -329,6 +329,13 @@ export const DetailsMap: React.FC<PropsType> = props => {
                 />
               </div>
             )}
+            {Number(props.viewPoints?.length) > 0 && (
+              <BackToMapButton
+                displayMap={props.displayMap}
+                setMapId={props.setMapId}
+                mapId={props.viewPoints?.[0].id}
```

Comment:
If `props.viewPoints` may be an empty Array, maybe:

```suggestion
                mapId={props.viewPoints?.[0]?.id}
```

Reply:
With or without optional chaining, searching for an `id` in an empty array returns `undefined`.

Reply:
It seems you're not getting `id` property from empty array, but from first cell which may be `undefined`.
If `props.viewPoints` is an empty Array, `props.viewPoints?.[0]` is `undefined`.
Getting `id` from `undefined` may throw.

Reply:
oh well :+1: